docs/root/api/grpc.rst

```diff
@@ -0,0 +1,144 @@
+type. Upon calling ``build()``, it returns a ``Request`` (the same type used for standard HTTP
+requests/streams) which is preconfigured for gRPC.
+
+To start a gRPC stream, create a ``Request`` using the ``GRPCRequestBuilder``.
+
+**Kotlin**::
+
+  val request = GRPCRequestBuilder("/pb.api.v1.Foo/GetBar", "api.envoyproxy.io", true)
+    .addHeader("x-custom-header", "foobar")
+    ...
+    .build()
+
+**Swift**::
+
+  let request = GRPCRequestBuilder(path: "/pb.api.v1.Foo/GetBar", authority: "api.envoyproxy.io", useHTTPS: true)
+    .addHeader(name: "x-custom-header", value: "foobar")
+    ...
+    .build()
+
+-----------------------
+``GRPCResponseHandler``
+-----------------------
+
+Very similarly to the HTTP ``ResponseHandler``, the ``GRPCResponseHandler`` allows for receiving
+updates to the gRPC stream and contains a set of callbacks that are called whenever an update
+occurs on the stream.
+
+This handler processes inbound gRPC responses, buffers data as necessary while chunks of
+protobuf messages are received, then finally passes fully formed protobuf data to the callbacks
+provided.
+
+Typically, consumers should listen to ``onMessage`` and use a protobuf library to deserialize
+the complete protobuf message data.
+
+**Kotlin**::
+
+  val handler = GRPCResponseHandler(Executor { })
+    .onHeaders { headers, grpcStatus, _ ->
+      Log.d("MainActivity", "Received gRPC status: " + statusCode + " and headers: " + headers)
+      Unit
+    }
+    .onMessage { messageData ->
+      // Deserialize message data here
+    }
+    .onTrailers { trailers ->
+      Log.d("MainActivity", "Trailers received: " + trailers)
+      Unit
+    }
+    .onError { error ->
+      Log.d("MainActivity", "Error received: " + error.message)
+      Unit
+    }
+    ...
+
+**Swift**::
+
+  let handler = GRPCResponseHandler()
+    .onHeaders { headers, grpcStatus, _ in
+      print("Received gRPC status: \(grpcStatus) and headers: \(headers)")
+    }
+    .onMessage { messageData in
+      // Deserialize message data here
+    }
+    .onTrailers { trailers in
+      print("Trailers received: \(trailers)")
+    }
+    .onError { error in
+      print("Error received: \(error.message)")
+    }
+    ...
+
+
+---------------------
+``GRPCStreamEmitter``
+---------------------
+
+Finally, a gRPC stream may be opened using a ``GRPCClient`` instance.
+
+Doing so returns a ``GRPCStreamEmitter`` which allows the sender to write to the stream,
+close the stream, etc.
+
+The ``sendMessage`` function should be invoked with the serialized data from a protobuf message.
+The emitter will then transform the provided data into the gRPC wire format and send it over the
+stream.
+
+**Kotlin**::
+
+  val envoy = AndroidEnvoyClientBuilder(...).build()
+  val grpcClient = GRPCClient(envoy)
+
+  val request = GRPCRequestBuilder(...).build()
+  val responseHandler = GRPCResponseHandler(...)
+  val grpcEmitter = grpcClient.send(request, responseHandler)
```

Comment:
For these examples, can we follow the builder chained-call pattern as opposed to capturing the `emitter` in a variable?

Reply:
Agree, specially since this is how we use it in code.

Reply:
Yep, updated. Actually realized that the gRPC emitters aren't chainable even though they should be. Updated the docs in this PR and am pushing functionality in https://github.com/lyft/envoy-mobile/pull/534